turbinia/evidence.py

```diff
@@ -342,14 +382,47 @@ def from_dict(cls, dictionary):
     new_object.__dict__.update(dictionary)
     return new_object
 
-  def serialize(self):
-    """Return JSON serializable object."""
+  def serialize_attribute(self, name: str) -> str:
+    """Returns JSON serialized attribute.
+    
+    Args:
+      name(str): the name of the attribute that will be serialized.
+    Returns:
+      A string containing the serialized attribute.
+    """
+    if hasattr(self, name):
+      try:
+        return json.dumps(getattr(self, name))
+      except (TypeError, OverflowError):
+        log.error(f'Attribute {name} in evidence {self.id} is not serializable')
+    else:
+      log.error(f'Evidence {self.id} has no attribute {name}')
+
+  def serialize(self, json_values: bool = False):
+    """Returns a JSON serialized object. The function will return A string
+    containing the serialized evidence_dict or a dict of serialized attributes
+    if json_values is true.
+    
+    Args:
+      json_values(bool): Returns only values of the dictionary as json strings
+        instead of the entire dictionary.
+
+    Returns:
+      JSON serialized object.
+    """
     # Clear any partition path_specs before serializing
     if hasattr(self, 'path_spec'):
       self.path_spec = None
-    serialized_evidence = self.__dict__.copy()
-    if self.parent_evidence:
-      serialized_evidence['parent_evidence'] = self.parent_evidence.serialize()
+    serialized_evidence = {}
+    if json_values:
+      for attribute_name in self.__dict__:
```

Comment:
This is going to result in having different sets of attributes per evidence type stored in Redis and the other objects that we store have a fixed list of attributes.  Is this a common way to use Redis?  Will this cause any issues when trying to use this data elsewhere (ie. trying to get attributes that may or may not be there)?  I wonder if we want to use the REQUIRED_ATTRIBUTES, or if that's too minimal for what we want here?

Reply:
```This is going to result in having different sets of attributes per evidence type stored in Redis and the other objects that we store have a fixed list of attributes.``` I would prefer other objects we store in Redis to have all their attributes stored as well (to be discussed? it's not entirely clear why currently only some attributes are persisted besides ones that are not JSON serializable). In this particular case, each evidence subclass could have additional attributes (required or not). Since the goal is to persist Evidence state in Redis, it makes sense to persist all JSON serializable attributes. In my view, it's cleaner to have a 1:1 correspondence between a particular Evidence object and it's stored metadata in Redis but I'm open to change this if we can optimize Redis keys if some attributes are only set at Runtime and may change every time the same Evidence object is instantiated.

```Is this a common way to use Redis?``` It's common practice to store JSON serialized objects in Redis either as strings or Redis hashes (key/value objects). 

```Will this cause any issues when trying to use this data elsewhere (ie. trying to get attributes that may or may not be there)? I wonder if we want to use the REQUIRED_ATTRIBUTES, or if that's too minimal for what we want here?``` We deserialize the Redis keys into objects and that'll include any/all attributes stored so it's a lot easier to avoid problems if we have all the attributes rather than a select few.